pe.c

```diff
@@ -1043,10 +1043,14 @@ handle_image (void *data, unsigned int datasize,
 				return EFI_UNSUPPORTED;
 			}
 
-			/* If it has nonzero size, and our bounds check made
-			 * sense, sizes match, then we believe it's okay. */
+			/* The virtual size corresponds to the size of the SBAT
+			 * metadata and isn't necessarily a multiple of the file
+			 * alignment. The on-disk size is a multiple of the file
+			 * alignment and is zero padded. Make sure that the
+			 * on-disk size is at least as large as virtual size,
+			 * and ignore the section if it isn't. */
 			if (Section->SizeOfRawData &&
-			    Section->SizeOfRawData == Section->Misc.VirtualSize &&
+			    Section->SizeOfRawData >= Section->Misc.VirtualSize &&
```

Comment:
Giant case of nit-picking:
You're right that we have to handle SizeOfRawData > VirtualSize, but I think we may actually need to allow SizeOfRawData <  VirtualSize as well in order to be correct - though probably we should handle that in our "end" computation.  From the PE spec:
|Offset|Size|Field|Description|
|--|--|--|--|
|8|4|VirtualSize|The total size of the section when loaded into memory. If this value is greater than SizeOfRawData, the  section is zero-padded. This field is valid only for executable images and should be set to zero for object files.|
|16|4|SizeOfRawData|The size of the section (for object files) or the size of the initialized data on disk (for image files).  or executable images, this must be a multiple of FileAlignment from the optional header. If this is less than VirtualSize,  the remainder of the section is zero-filled. Because the SizeOfRawData field is rounded but the VirtualSize field is not, it is possible for SizeOfRawData to be greater than VirtualSize as well. When a section contains only uninitialized data, this field should be zero.|

Just so I don't have to have a page of data in my examples, let's pretend FileAlignment is 7 bytes.  As I read it, in the case someone makes the very minor and understandable error of including a .sbat section that is, say: "shim,0\n\000", the compiler is allowed to give us a binary where SizeOfRawData == 7 and VirtualSize == 8.  In this case we should be padding out the section after loading it, though of course for .sbat validation nothing in practice will change, because we're not going to interpret all the "\000" padding anyway.

I don't think this needs to hold back this patch, and I've filed it as https://github.com/rhboot/shim/issues/287